src/main/java/duke/commands/Deadline.java

```diff
@@ -0,0 +1,33 @@
+package duke.commands;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDateTime by;
+
+    public Deadline(String description, LocalDateTime by) {
+        super(description);
+        this.by = by;
+    }
+
+    public Deadline(String description, boolean isDone, LocalDateTime by) {
+        super(description, isDone);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+        String dateString = by.format(format);
+        return "[D]" + super.toString() + " (by: " + dateString + ")";
+    }
+
+    @Override
+    public String printFormat() {
```

Comment:
Should this method be renamed to something like toPrintFormat(), so that it sounds more like a verb? The same applies to the printFormat() method in the Event, Task and Todo classes.